package/kedro_viz/api/graphql.py

```diff
@@ -135,22 +136,42 @@ def format_run_tracking_data(tracking_data: Dict) -> JSONObject:
     """
     formatted_tracking_data = defaultdict(list)
 
-    for run_id, run_tracking_data in tracking_data.items():
-        for tracking_name, data in run_tracking_data.items():
-            formatted_tracking_data[tracking_name].append(
-                {"runId": run_id, "value": data}
-            )
+    tracking_keys = []
```

Comment:
I would suggest doing this a bit differently:
* leave the stuff that just generates `formatted_tracking_data` as it was before
* this should include all the metrics (both common across runs and not common across runs)
* _then_, if  `show_diff = False`, go through `formatted_tracking_data` and remove the ones where it's not common across all runs
* return whatever is left in `formatted_tracking_data`

It might look something like this (completely untested...):
 ```suggestion
 # Same as before:
formatted_tracking_data = defaultdict(list)
for tracking_name, data in run_tracking_data.items():
    formatted_tracking_data[tracking_name].append({"runId": run_id, "value": data})
    
# New stuff:
if show_diff:
     for tracking_name, all_run_data in list(formatted_tracking_data.items()):
         if any(one_run_data["value"] is None for one_run_data in all_run_data):
              del formatted_tracking_data[tracking_name]
```

Reply:
Thanks, this is smarter. But this doesn't handle the case where show_diff = false and we need to explicitly add 'null' to the keys which have no value right? Also Merel's question is should we do that in the backend?

Reply:
Ohh, I thought the original `formatted_tracking_data` already contained those nulls and then what you were trying to do is remove them if `show_diff=True`. But you're saying the original `formatted_tracking_data` didn't contain the nulls and if `show_diff=False` you need to add them?

Reply:
Also, just wondering... this kind of dictionary reshaping thing might be much easier to do in a pandas DataFrame. I see pandas is already a dependency of kedro-viz but don't know if you use it for data manipulation anywhere or it's preferred to do all this sort of thing with dictionaries?